gadgets/deadlock/go/program.go

```diff
@@ -264,13 +267,33 @@ func gadgetInit() int {
 		return nil
 	}, 0)
 
-	// Clean up dead processes using pid from the `process_exit` datasource
+	// Clean up remains of dead processes using pid from the `process_exit` datasource
 	dsProcessExit.Subscribe(func(source api.DataSource, data api.Data) {
 		deadPid, err := deadPidF.Uint32(data)
 		if err != nil {
 			api.Warnf("failed to get pid: %s", err)
 			return
 		}
+
+		// Delete dead keys from the `thread_to_held_mutexes` BPF map
+		if g, exists := graphs[deadPid]; exists {
+			deadTids := make(map[uint32]struct{})
+			edges := g.Edges()
+			for _, edge := range edges {
+				// Built set of TIDs to delete
+				tid := g.Attributes(edge[0], edge[1])["tid"].(uint32)
+				deadTids[tid] = struct{}{}
+			}
+			for deadTid := range deadTids {
+				err := threadToHeldMutexesBPFMap.Delete(deadTid)
+				if err != nil {
+					api.Warnf("failed to delete key from threads: %s", err)
+					continue
+				}
+			}
+		}
```

Comment:
This can be further optimized by maintaining a separate map in Go that maps PIDs to its threads so as to avoid traversing all edges of the graph, but maybe I'm just overthinking and this would work just fine without having to worry about ANOTHER buffer. WDYT?